pages/docs/reference/js-ir-compiler.md

```diff
@@ -0,0 +1,71 @@
+and tooling features described in this document are subject to change in future Kotlin versions.
+{:.note}
+
+The (currently still experimental) Kotlin/JS IR compiler back-end is the main focus of innovation around Kotlin/JS, and paves the way forward for the technology. 
+
+Rather than directly generating JavaScript code from Kotlin source code, the Kotlin/JS IR compiler back-end leverages a new approach. Kotlin source code is first transformed into an intermediate representation (IR), which is subsequently compiled into JavaScript. This allows for more aggressive optimizations, and enables improvements on pain points that were present in the previous compiler. This includes the areas of developer experience, generated code size (dead code elimination), and JavaScript and TypeScript ecosystem interoperability, to name just a few.
+
+The experimental IR compiler back-end is available starting with Kotlin 1.4 through the Kotlin/JS Gradle plugin. To enable it in your project, pass a compiler type to the `js` function in your Gradle build script:
+
+<!--suppress ALL -->
+<div class="sample" markdown="1" mode="groovy" theme="idea">
+
+```groovy
+kotlin {
+    js(IR) { // or: LEGACY, BOTH
+        // . . .
+    }
+    binaries.executable()
+}
+```
+
+</div>
+
+- `IR` uses the new IR compiler back-end for Kotlin/JS.
+- `LEGACY` uses the default compiler back-end.
+- `BOTH` compiles your project with the new IR compiler as well as the default compiler back-end. This is mainly useful for authoring libraries that are compatible with both back-ends, see [below](#authoring-libraries-for-the-ir-compiler-with-backwards-compatibility).
+
+The compiler type can also be set in the `gradle.properties` file, with the key `kotlin.js.compiler=ir`. (This behaviour is overwritten by any settings in the `build.gradle(.kts)`, however).
+
+## Current limitations of the IR compiler
+
+A major change with the new IR compiler back-end is the **absence of binary compatibility** with the default back-end. A lack of such compatibility between the two back-ends for Kotlin/JS means that a library created with the new IR compiler back-end can’t be used from the default back-end, and vice versa.
+
+If you want to use the IR compiler back-end for your project, you need to **update all Kotlin dependencies to versions that support this new back-end**. Libraries published by JetBrains for Kotlin 1.4+ targeting Kotlin/JS already contain all artifacts required for usage with the new IR compiler back-end.
+
+**If you are a library author** looking to provide compatibility with the current compiler back-end as well as the new IR compiler back-end, additionally check out the [“Authoring libraries for the IR compiler”](#authoring-libraries-for-the-ir-compiler-with-backwards-compatibility) section.
+
+The IR compiler back-end also has some discrepancies in comparison to the default back-end. When trying out the new back-end, it's good to be mindful of these possible pitfalls.
```

Comment:
Maybe it is worth adding a note and/or a link to related issues for first and second points?

Reply:
I've added links to:

https://youtrack.jetbrains.com/issue/KT-39447
and
https://youtrack.jetbrains.com/issue/KT-40525

If the issue IDs should be changed, please let me know.

Reply:
lgtm